src-java/base-topology/base-storm-topology/src/test/java/org/openkilda/wfm/share/mappers/HaFlowHistoryMapperTest.java

```diff
@@ -452,8 +458,47 @@ public void createHaFlowDump() {
         assertEquals(source.getReversePath().getStatus(), dump.getReversePath().getStatus());
     }
 
-    private HaFlowEventDump createHaFlowEventDump(org.openkilda.model.history.DumpType dumpType, String correlationId) {
-        return new HaFlowEventDump(HaFlowEventDumpDataImpl.builder()
+    @Test
+    public void whenDumpContainsEmptyListsInPathPayload_theWholePathPayloadIsExcluded() throws JsonProcessingException {
+        HaFlowDumpPayload payloadWithEmptyPaths = new HaFlowDumpPayload("1", null, null, null, null, null, null, null,
+                null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null,
+                null, null, null, new HaFlowPathPayload(null, null, null, null, null, null, null, null, null, null,
+                null, null, Collections.EMPTY_LIST, Collections.EMPTY_LIST), new HaFlowPathPayload(null, null, null,
+                null, null, null, null, null, null, null, null, null, Collections.EMPTY_LIST, Collections.EMPTY_LIST));
+
+        ObjectMapper objectMapper = new ObjectMapper();
+
+        String json = objectMapper.writeValueAsString(payloadWithEmptyPaths);
+        assertFalse(json.contains("protected_reverse_path"),
+                "Protected forward must not be present, because all fields are null and lists are empty");
+        assertFalse(json.contains("protected_forward_path"),
+                "Protected forward must not be present, because all fields are null and lists are empty");
+    }
+
+    @Test
+    void whenDumpContainsNonEmptyListsInPathPayload_theWholePathPayloadIsIncluded() throws JsonProcessingException {
+        HaFlowDumpPayload payloadWithPaths = new HaFlowDumpPayload("1", null, null, null, null, null, null, null, null,
```

Comment:
I suggest to move creation of totally empty  `HaFlowDumpPayload` to the method and modify needed values in this test. 
Also, Would it play trick if you add @NoArgsConstructor for HaFlowDumpPayload class?